### PR TITLE
Implementation of the algorithm for the Koch snowflake

### PR DESCRIPTION
**Pull request by algobytewise**
Implementation of the algorithm for the Koch snowflake

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by algobytewise** [comment]
Hello, it would be great if I could get some feedback on how to respond to the failed tests.
(1) algorithms-keeper complains that VECTOR_1, VECTOR_2 and VECTOR_1 are not snake_case. But they are intended as constants, not as variables.
(2) I'm not sure what doctest to implement for the plot-function. It is not part of the algorithm, it just draws a graph and it has no return value.

**Comment by mrmaxguns** [comment]
The bot just gives guidance. Tests failed because you didn't run isort. Here's the isort diff:
```
 from __future__ import annotations
-import numpy
 
+import numpy
 
 # initial triangle of Koch snowflake
 VECTOR_1 = numpy.array([0, 0])
```
As you can see, a newline was added between `from __future__ import annotations` and `import numpy`. The constants are fine. For the tests, you should probably add a small note in the relevant function why you chose not to include tests, so that future contributors know.

**Comment by algobytewise** [comment]
Dear mrmaxguns, thank you for the feedback, I have implemented the changes you suggested.

**Comment by cclauss** [comment]
This looks good.  Nice work.  I am not a fan of uncategorized algorithms in the `other` directory because it makes it harder for visitors to find algorithms of interest.  Would it make sense to move this to the `cellular_automata` or `graphics` directory instead?

File: other/koch_snowflake.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    axes.set_aspect("equal")
+
+    # matplotlib.pyplot.plot takes a list of all x-coordinates and a list of all
+    # y-coordinates as inputs, which need to be constructed from our vector-list
+    x_coordinates = []
+    for vector in vectors:
+        x_coordinates.append(vector[0])
+    y_coordinates = []
+    for vector in vectors:
+        y_coordinates.append(vector[1])
```
Comment (cclauss) [inline other/koch_snowflake.py:111]: Please make these `list comprehensions`.

File: other/koch_snowflake.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    Loops through each pair of adjacent vectors. Each line between two adjacent
+    vectors is divided into 4 segments by adding 3 additional vectors in-between
+    the original two vectors. The vector in the middle is constructed through a
+    60 degree rotation so it is bent outwards.
+    >>> iteration_step([numpy.array([0, 0]), numpy.array([1, 0])])
+    [array([0, 0]), array([0.33333333, 0.        ]), array([0.5       , \
+0.28867513]), array([0.66666667, 0.        ]), array([1, 0])]
+    """
+    new_vectors = []
+    for i in range(len(vectors) - 1):
```
Comment (cclauss) [inline other/koch_snowflake.py:64]: `range(len())` is almost always a sign that `enumerate()` could be used instead.

File: other/koch_snowflake.py (hunk 1 above)
Comment (cclauss) [reply other/koch_snowflake.py:111]: Or maybe even slicker, use `zip()` instead.

**Comment by algobytewise** [comment]
I have implemented the changes, thanks for the advice on better looping. I think `graphics` is a better fit so I've put it there. My preferred solution would be to have a new directory for `fractals` and move it there together with `other/sierpinski_triangle.py`.

**Review by cclauss — APPROVED** [review]
LGTM

**Pull request merged**